apps/sage_intacct/utils.py

```
@@ -1328,151 +1329,266 @@ def __construct_charge_card_transaction(
-                    'COSTTYPEID': 'cost_type_id',
-                    'PROJECTID': 'project_id'
-                }
+            allocation_detail = DestinationAttribute.objects.filter(
+                workspace_id=workspace_id,
+                attribute_type='ALLOCATION',
+                value=lineitem.allocation_id
+            ).first().detail
 
-                allocation_detail = DestinationAttribute.objects.filter(workspace_id=self.workspace_id, attribute_type='ALLOCATION', value=lineitem.allocation_id).first().detail
+            for allocation_dimension, dimension_variable_name in allocation_mapping.items():
+                if allocation_dimension in allocation_detail.keys():
```

Ruff (0.8.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)